client/remoterepo.c

```diff
@@ -986,6 +1015,49 @@ TDNFDownloadPackageToCache(
                                 &pszNormalRpmCacheDir);
     BAIL_ON_TDNF_ERROR(dwError);
 
+    dwError = TDNFDownloadPackageToTree(pTdnf,
+                                        pszPackageLocation,
+                                        pszPkgName,
+                                        pszRepoName,
+                                        pszNormalRpmCacheDir,
+                                        ppszFilePath);
+    BAIL_ON_TDNF_ERROR(dwError);
+cleanup:
+    TDNF_SAFE_FREE_MEMORY(pszNormalRpmCacheDir);
+    TDNF_SAFE_FREE_MEMORY(pszRpmCacheDir);
+    return dwError;
+error:
+    goto cleanup;
+}
+
+uint32_t
+TDNFDownloadPackageToTree(
```

Comment:
What is the main difference between TDNFDownloadPackageToTree() and the existing TDNFDownloadPackageToDirectory() function? Both have same args and calls TDNFDownloadPackage().
Names are confusing, if you could add some comments about these functions it would be easy to understand.

Reply:
Good point. I have added comments.